### DISCUSSION TITLE
Standard way to pop App Notification using tm-notify

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#7324] [Ideas]
Some users ask is there a repeat task reminder, and want app level notification.

In Tiddloid and Tidgi we can use a plugin to hook to `tm-notify`, and pop app level OS native notification. Also in web platform, we can pop browser notification in some way.

I suggest add a new parameter to `tm-notify`, to ask app or browser to trigger native notification. This can be implemented as plugin, but needs a standard.

**Comment by Jermolene** [comment]
Thanks @linonetwo

I think that many users would be satisfied if they were able to choose between two simple options:

* All notifications are handled within TW
* All notifications are passed to the native notification mechanism

The finer degree of control of being able to selectively route notifications sounds useful, but I think that users are better placed to make the choice than the developers who create the wikitext that generate the notifications. One could imagine a user interface that lists all the notification tiddlers, and provides a means to selecting how it should be routed.

**Reply by pmario** [reply]
> I think that users are better placed to make the choice than the developers who create the wikitext that generate the notifications.

+1 for this one. In the past I did "always block" background tasks and web notifications if any website asks for it. I just don't want it.

If I need system notifications I use my self hosted calendar which syncs with all my devices. Having a possibility to communicate with them would be cool ... because it would be cool ;)

**Reply by linonetwo** [reply]
Get it, I will hook into this message to transform all in-wiki notification to native notification. And TidGi already has "mute notification" feature. I just need to remove unnecessary in-wiki notification that popup everytime it do a backup.
